### DISCUSSION TITLE
Is sending unprepared statements possible when using $queryRaw/$executeRaw?

### DISCUSSION THREAD

**Discussion by kldzj** [prisma/prisma#3931] [Q&A] [answered]
I need to lock the table I'm currently working on, but the Prisma Client seems to always send prepared statements (even when not required), therefore I cannot lock the table using Prisma.

Is there a way I can force the client to send a statement that will not be prepared?

I initially thought not using tagged-templates and not passing any variables would achieve that, but this still errors: `await prisma.$executeRaw('LOCK TABLES table_name WRITE')`

**Comment by ryands17** [comment, marked as answer]
I suppose using the function equivalent will not contain prepared statements.

For e.g.

```ts
prisma.$queryRaw(`select * from "User" where id = ${1}`)
```

This shouldn't be a PS.

**Reply by kldzj** [reply]
Thanks for your quick response! I've tried that, as I thought the docs stated exactly this, but it still seems to send prepared statements when normally calling the queryRaw/executeRaw method.

These calls:
```js
await prisma.$executeRaw('LOCK TABLES table_name WRITE')
```

```js
await ctx.db.$executeRaw('UNLOCK TABLES')
```

still throw the error `This command is not supported in the prepared statement protocol yet`

**Reply by ryands17** [reply]
@kldzj 👋 
Yes I checked and that returns one too. Currently there's no API in Prisma that offeres unprepared statements unfortunately.

**Reply by kldzj** [reply]
I see this is currently tracked in #3337, thank you very much for clarifying Ryan.

**Reply by ryands17** [reply]
Hey @kldzj 👋 
If you could spare some time here, would it be possible for you to explain your use case for adding locks on tables?

**Reply by kldzj** [reply]
Hey @ryands17, we're manipulating a nested set structured table, which could break if another query would alter results before another longer running statement.

**Reply by ryands17** [reply]
Do you have any sample SQL statements so that we can use those to test it from our end as well?

**Reply by kldzj** [reply]
@ryands17 if you need anything else to reproduce let me know.

```sql
# Create Table
CREATE TABLE `navigation_tree` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `name` varchar(255) NOT NULL,
  `tree_left` int(11) DEFAULT NULL,
  `tree_right` int(11) DEFAULT NULL,
  `tree_level` int(11) DEFAULT NULL,
  PRIMARY KEY (`id`)
);

# Plant tree
INSERT INTO
  navigation_tree (name, tree_left, tree_right, tree_level)
VALUES
  ('Root', 1, 32, 0),
  ('A', 2, 3, 1),
  ('B', 4, 11, 1),
  ('B1', 5, 6, 2),
  ('B2', 7, 8, 2),
  ('B3', 9, 10, 2),
  ('C', 12, 17, 1),
  ('C1', 13, 14, 2),
  ('C2', 15, 16, 2),
  ('D', 18, 19, 1),
  ('E', 20, 31, 1),
  ('E1', 21, 22, 2),
  ('E2', 23, 28, 2),
  ('E2-1', 24, 25, 3),
  ('E2-2', 26, 27, 3),
  ('E3', 29, 30, 2);
```

```js
// Mutate
// Values used in the statements always come from the node that is being deleted/added to.
// e.g. SELECT tree_right, tree_level FROM navigation_tree WHERE name = "E2";

// a) Grow
$executeRaw('LOCK TABLES navigation_tree WRITE')
$executeRaw`UPDATE navigation_tree SET tree_left = tree_left + 2 WHERE tree_left > ${right}`
$executeRaw`UPDATE navigation_tree SET tree_right = tree_right + 2 WHERE tree_right >= ${right}`
$executeRaw`INSERT INTO navigation_tree (name, tree_left, tree_right, tree_level)
                      VALUES (${name}, ${right}, ${right + 1}, ${level + 1})`
$executeRaw('UNLOCK TABLES')

// b) Shrink
$executeRaw('LOCK TABLES navigation_tree WRITE')
$executeRaw`DELETE FROM navigation_tree WHERE tree_left BETWEEN ${tree_left} AND ${tree_right}`
$executeRaw`UPDATE navigation_tree SET tree_left = tree_left - 2 WHERE tree_left > ${tree_right}`
$executeRaw`UPDATE navigation_tree SET tree_right = tree_right - 2 WHERE tree_right > ${tree_right}`
$executeRaw('UNLOCK TABLES')
```

**Reply by ryands17** [reply]
Thanks a lot for this :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
